internal/events/pubsub.go

```diff
@@ -3,48 +3,29 @@ package events
 import (
 	gcppubsub "cloud.google.com/go/pubsub"
 	"context"
-	"github.com/googleapis/google-cloudevents-go/cloud/auditdata"
 	"go.uber.org/zap"
-	"google.golang.org/protobuf/encoding/protojson"
 )
 
-type notifier struct {
-	t   Subscription
+type subscription struct {
+	t   *gcppubsub.Subscription
 	log *zap.SugaredLogger
 }
 
 type Subscription interface {
-	Receive(ctx context.Context, f func(context.Context, *gcppubsub.Message)) error
+	Receive(ctx context.Context, event chan<- *gcppubsub.Message)
 }
 
-func (n *notifier) Receive(ctx context.Context, event chan<- InterruptionEvent) {
-	err := n.t.Receive(ctx, func(ctx context.Context, m *gcppubsub.Message) {
+func (s *subscription) Receive(ctx context.Context, event chan<- *gcppubsub.Message) {
+	err := s.t.Receive(ctx, func(ctx context.Context, m *gcppubsub.Message) {
 		m.Ack()
-		e, err := formatInterruptionEventFromPubSubMessage(m)
-		if err != nil {
-			n.log.Errorf("failed to handle event: %s", err.Error())
-			return
-		}
-		event <- e
+		event <- m
 	})
 	if err != nil {
-		n.log.Error(err)
+		s.log.Error(err)
 	}
 	close(event)
 }
 
-func formatInterruptionEventFromPubSubMessage(m *gcppubsub.Message) (InterruptionEvent, error) {
-	entry := auditdata.LogEntryData{}
-	err := protojson.UnmarshalOptions{DiscardUnknown: true}.Unmarshal(m.Data, &entry)
-	if err != nil {
-		return InterruptionEvent{}, err
-	}
-	return InterruptionEvent{
-		MessageID:  m.ID,
-		ResourceID: entry.ProtoPayload.ResourceName,
-	}, nil
-}
-
 // NewPubSubNotifierInput defines all required fields to create a PubSubNotifier
 type NewPubSubNotifierInput struct {
```

Comment:
nit (since you're not editing it here): I think `PubSubNotifierOpts` is a more idiomatic name